stdlib/LinearAlgebra/src/ldlt.jl

```diff
@@ -48,13 +48,13 @@ julia> S
   ⋅        0.545455  3.90909
 ```
 """
-function ldlt!(S::SymTridiagonal{T,V}) where {T<:Real,V}
+function ldlt!(S::SymTridiagonal{T,V}) where {T,V}
     n = size(S,1)
     d = S.dv
     e = S.ev
     @inbounds @simd for i = 1:n-1
         e[i] /= d[i]
-        d[i+1] -= abs2(e[i])*d[i]
+        d[i+1] -= e[i]^2*d[i]
```

Comment:
Was `abs2` incorrect here?

Reply:
Yes, but the previous version was also restricted to `Real` so it didn't matter. Had the type been HermitianTridiagonal then `abs2` would have been correct.